lib/vagrant-libvirt/action/package_domain.rb

```diff
@@ -32,6 +32,8 @@ def call(env)
           )
           domain = env[:machine].provider.driver.connection.servers.get(env[:machine].id.to_s)
           root_disk = domain.volumes.select do |x|
+            !x.nil?
+          end.select do |x|
```

Comment:
no need to split into 2 separate blocks: `!x.nil? && x.name == libvirt_domain.name + '.img'` should suffice as ruby does greedy conditional evaluation.

Would you mind duplicating the test at https://github.com/vagrant-libvirt/vagrant-libvirt/blob/2e8df0fa1b346d15aa728162c1cf7ed4475fb94e/spec/unit/action/package_domain_spec.rb#L37-L62 and change the line returning the volumes to return a nil object first.

With a little bit of pruning, should just be a case of adding the following in:
```
    context 'with nil volume' do
      let(:root_disk) { double('libvirt_domain_disk') }
      before do
        allow(root_disk).to receive(:name).and_return('default_domain.img')
        allow(domain).to receive(:volumes).and_return([nil, root_disk])
        allow(libvirt_domain).to receive(:name).and_return('default_domain')
        allow(subject).to receive(:download_image).and_return(true)
      end

      it 'should succeed' do
        expect(subject).to receive(:`).with(/qemu-img info .*\/box.img | grep 'backing file:' | cut -d ':' -f2/).and_return("some image")
        expect(subject).to receive(:`).with(/qemu-img rebase -p -b "" .*\/box.img/)
        expect(subject).to receive(:`).with(/virt-sysprep --no-logfile --operations .* -a .*\/box.img .*/)
        expect(subject).to receive(:`).with(/virt-sparsify --in-place .*\/box.img/)
        expect(subject).to receive(:`).with(/qemu-img info --output=json .*\/box.img/).and_return(
          { 'virtual-size': 5*1024*1024*1024 }.to_json
        )

        expect(subject.call(env)).to be_nil
      end
    end
```

Reply:
Did that. Do you want me to squash?

Reply:
Think it should be fine given the two commits are relatively straightforward